src/main/java/org/assertj/core/error/ShouldHaveSuperclass.java

```diff
@@ -37,11 +42,19 @@
    * @return the created {@code ErrorMessageFactory}.
    */
   public static ErrorMessageFactory shouldHaveSuperclass(Class<?> actual, Class<?> superclass) {
-    return new ShouldHaveSuperclass(actual, superclass);
+    Class<?> actualSuperclass = actual.getSuperclass();
+    if (actualSuperclass == null) {
+      return new ShouldHaveSuperclass(actual, superclass);
```

Comment:
is it possible? I think we are checking that `superClass` is not null in `hasSuperClass` so we should not need this one